src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -4,12 +4,40 @@
 
 public class StorageImpl<K, V> implements Storage<K, V> {
 
+    public static final int LENGTH = 10;
+    private K[] keyArray;
+    private V[] valueArray;
+    private int count = 0;
+    private int encounter = 0;
+
+    public StorageImpl() {
+        this.keyArray = (K[]) new Object[LENGTH];
+        this.valueArray = (V[]) new Object[LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < count; i++) {
+            if (key == keyArray[i]) {
```

Comment:
@EliosPride it shouldn't work in case when key is object. Please fix and add additional test to cover this case.